### PR TITLE
Adds support for host to know if firmware is using options tied to th…

### PR DESCRIPTION
**Pull request by mjstn**
Enables host to see hardware options from a new message from firmware.  
Host can now detect option that hardware is using 6-state encoders and then this host code adjusts odometry for proper tics to radians values so that odometry is correct.
Lastly we change the warning for unrecognized message from firmware so it is now a ROS_WARN_ONCE.  Meant to do this last time but did not do it then.
This host code will support legacy firmware or newer 6-state encoder enabled firmware.

### CODE REVIEW

File: include/ubiquity_motor/motor_parmeters.h [hunk 1]
```diff
@@ -116,6 +119,8 @@ struct FirmwareParams {
             nh, "ubiquity_motor/fw_max_pwm", max_pwm);
         deadman_timer = getParamOrDefault(
             nh, "ubiquity_motor/deadman_timer", deadman_timer);
+        hw_options = getParamOrDefault(
```
Comment (rohbotics) [inline include/ubiquity_motor/motor_parmeters.h]: I strongly object to this being a parameter, it should *only* ever be read from the firmware

File: src/motor_hardware.cc [hunk 2]
```diff
@@ -38,11 +38,11 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
 // 17.2328767123
 // gear ratio of 4.29411764706:1
-#define TICS_PER_RADIAN (41.0058030317 / 2)
-#define QTICS_PER_RADIAN (TICS_PER_RADIAN * 4)
+#define TICS_PER_RADIAN_ENC_3_STATE (41.0058030317/2)
+#define QTICS_PER_RADIAN_ENC_3_STATE (tics_per_radian*4)
 #define VELOCITY_READ_PER_SECOND \
     10.0  // read = ticks / (100 ms), so we have scale of 10 for ticks/second
-#define CURRENT_FIRMWARE_VERSION 24
+#define LOWEST_FIRMWARE_VERSION 28
```
Comment (rohbotics) [inline src/motor_hardware.cc:49]: Thanks for updating this

**Review by rohbotics — CHANGES_REQUESTED** [review]

File: include/ubiquity_motor/motor_parmeters.h (hunk 1 above)
Comment (mjstn) [reply include/ubiquity_motor/motor_parmeters.h]: Agreed.  Not sure why I had it as settable from ROS side but at the time I was thinking we may want to make it switchable in some way which we discussed. I had not gone back and removed these 2 lines after that.   It does seem that it is a sort of read only member of FirmwareParams and so I leave it there so it is tightly bound to firmware, that part seems right.  The option we had discussed on the phone was if we ever wanted to try to make it switchable and I am glad you agreed we do not as that would be problematic.

**Comment by rohbotics** [comment]
@mjstn Is this ready for re-review?

**Review by jim-v — APPROVED** [review]
:+1:

**Review by rohbotics — APPROVED** [review]

**Pull request merged**